src/ctia/task/migration/migrate_es_stores.clj

```diff
@@ -75,6 +77,109 @@
   (let [coercer (coerce-to-fn Model)]
     #(reduce (partial append-coerced coercer) {} %)))
 
+(s/defschema BatchParams
+  (st/optional-keys
+   {:source-store StoreMap
+    :target-store StoreMap
+    :migrated-count s/Int
+    :buffer-size s/Int
+    :search_after [s/Any]
+    :migrations s/Any
+    :entity-type s/Keyword
+    :batch-size s/Int
+    :migration-id s/Str
+    :list-coerce s/Any
+    :migration-es-conn ESConn
+    :confirm? s/Bool
+    :documents [{s/Any s/Any}]
+    :query ESQuery}))
+
+
+(s/defn read-source :- (s/maybe BatchParams)
+  "This function retrieves in `source-store`a batch of documents that match the given `query`.
+   When not nil, the `search_after` parameter is used to skip previously retrieved data. The
+   returned result prepares the next batch parameters with new `search_after` along with the
+   documents that have to be written in target."
+  [{:keys [source-store
+           search_after
+           batch-size
+           query]
+    :as batch-params} :- (s/maybe BatchParams)]
+    ;; loop that reads batches from source and produces them in chan
+  (when batch-params
+    (let [{:keys [data paging]} (mst/query-fetch-batch query
+                                                       source-store
+                                                       batch-size
+                                                       0
+                                                       "asc"
+                                                       search_after)
+          next-search-after (:sort paging)]
+      (when (seq data)
+        (assoc batch-params
+               :documents data
+               :search_after next-search-after)))))
+
+(s/defn write-target :- s/Int
+  "This function writes a batch of documents which are (1) modified with `migrations` functions,
+   (2) validated by `list-coerce` and (3) written into given `target-store`. This function updates
+   the number of successfully migrated documents and search_after in the migration state identified
+   by given`migration-id` It finally returns this new number of successfully migrated documents."
+  [migrated-count :- s/Int
+   {:keys [target-store
+           documents
+           search_after
+           migrations
+           entity-type
+           batch-size
+           migration-id
+           list-coerce
+           migration-es-conn
+           confirm?]
+    :as batch-params} :- BatchParams]
+  (let [migrated (transduce migrations conj documents)
+        {:keys [data errors]} (list-coerce migrated)
+        new-migrated-count (+ migrated-count (count data))]
+    (doseq [entity errors]
+      (let [message
+            (format "%s - Cannot migrate entity: %s"
+                    (name entity-type)
+                    (pr-str entity))]
+        (log/error message)))
+    (when confirm?
+      (when (seq data) (mst/store-batch target-store data))
+      (mst/rollover target-store batch-size new-migrated-count)
```

Comment:
I don't see the usage of `batch-size` and `new-migrated-count` in the `rollover` fn

Reply:
blessed catch!